### DISCUSSION TITLE
Is there an obvious way to check if dialogue is currently being displayed?

### DISCUSSION THREAD

**Discussion by KingSnailtan** [nathanhoad/godot_dialogue_manager#184] [Q&A]
Hello! This is probably a very simple problem, but I thought I might ask for some tips.
So basically the character can still move when dialogue is displayed, and I want to disable that when the dialogue box is open.

Is there like a global variable that I can read, or events or something (like dialgue_start / end" etc) or do I have to come op with something of my own. Its easy to just lock controls when I start dialogue as I do this in my own logic anyway, but I have no obvious way to check if dialogue is finshed.

I could add methods to every dialogue using the yknow function calls you can add to your dialgogue, but that seems a bit counter intuitive. Am I missing something? thanks :D

**Comment by KingSnailtan** [comment]
I just saw that the autoload singleton has a "dialogue_ended" signal, that should do it, but it doesnt. It doesnt seem to fire when the dialogue ends.

**Comment by nathanhoad** [comment]
If you handle player movement via the `_unhandled_input` hook then they will stop accepting input when the dialogue balloon is present. There's an example of this in the [Beginner's Dialogue repository](https://github.com/nathanhoad/beginner_godot4_dialogue/blob/finished/characters/coco/coco.gd#L17)

**Reply by jdclifton2** [reply]
Hi,

Is there any other way to do this than doing player movement in _unhandled_input? I calculate my input vector in my physics_process, and it doesn't really make sense to me to put it in _unhandled_input.

**Reply by nathanhoad** [reply]
The input vector should be calculated in `_unhandled_input` and then processed in `_physics_process` otherwise you'll end up making things difficult for yourself down the track (and not just with dialogue handling).

If you're sure you want to do it differently then you'll have to keep track of a variable somewhere that allows/prevents player movement and then have a listener on [the `dialogue_ended` signal](https://github.com/nathanhoad/godot_dialogue_manager/blob/main/docs/API.md#dialoguemanager) that the Dialogue Manager emits.

**Reply by jdclifton2** [reply]
Okay, thanks for the heads up. I will take your word for it and follow your recommendations. Thanks for the excellent library, I am finding it intuitive and enjoyable to use :).
